### ISSUE TITLE
jenkins_plugin cannot install the plugin and giving "Cannot get list of plugins" error

### ISSUE THREAD

**Issue by arbabnazar** [ansible/ansible-modules-extras#2953]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

jenkins_plugin
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel e07b854505) last updated 2016/09/17 07:14:35 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 2b9300ec28) last updated 2016/09/17 07:15:26 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 35210ea450) last updated 2016/09/17 07:15:30 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

- Ubuntu 14.04 LTS (Remote Server)
- Mac OS  X 10.11.6 (Control Machine)
##### SUMMARY

jenkins_plugin : Cannot get list of plugins
##### STEPS TO REPRODUCE

Here is the snippet of code that generator this error:

``` yaml
- jenkins_plugin:
    name: greenballs
```
##### EXPECTED RESULTS

It should install the `greenballs` plugins on jenkins
##### ACTUAL RESULTS

```
fatal: [192.168.33.110]: FAILED! => {
    "changed": false,
    "details": "HTTP Error 500: Server Error",
    "failed": true,
    "invocation": {
        "module_args": {
            "backup": null,
            "content": null,
            "delimiter": null,
            "directory_mode": null,
            "follow": false,
            "force": false,
            "force_basic_auth": true,
            "group": "jenkins",
            "http_agent": "ansible-httpget",
            "jenkins_home": "/var/lib/jenkins",
            "mode": "0644",
            "name": "greenballs",
            "owner": "jenkins",
            "params": null,
            "regexp": null,
            "remote_src": null,
            "selevel": null,
            "serole": null,
            "setype": null,
            "seuser": null,
            "src": null,
            "state": "present",
            "timeout": 30.0,
            "unsafe_writes": null,
            "updates_expiration": 86400,
            "updates_url": "https://updates.jenkins-ci.org",
            "url": "http://localhost:8080",
            "url_password": null,
            "url_username": null,
            "use_proxy": true,
            "validate_certs": true,
            "version": null,
            "with_dependencies": true
        },
        "module_name": "jenkins_plugin"
    },
    "msg": "Cannot get list of plugins"
}
```

@jtyr

**Comment by ansibot** [comment]
@jtyr ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jtyr** [comment]
Thanks for your report, @arbabnazar. Please try to authenticate as admin or an user with admin rights.

**Comment by arbabnazar** [comment]
@jtyr can you please provide the example of it, I am really new to this plugin. Thanks

**Comment by jtyr** [comment]
Sure. Please have a look at the following example taken from the module [documentation](http://docs.ansible.com/ansible/jenkins_plugin_module.html):

```
- name: My Jenkins play
  vars:
    # Variable holding the login credentials
    my_jenkins_params:
      url_username: admin
      url_password: p4ssw0rd
      url: http://localhost:8080
  tasks:
    - name: Install plugin
      jenkins_plugin:
        name: build-pipeline-plugin
        # This is where we use the above variable
        params: "{{ my_jenkins_params }}"
```

**Comment by jtyr** [comment]
Did it work for you, @arbabnazar? If yes, please feel free close the issue.

**Comment by ansibot** [comment]
@jtyr, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jtyr** [comment]
I believe this is `notabug`. Please feel free to re-open the issue if there is still some problem.

**Comment by uddhavb** [comment]
i am getting this failure message:

FAILED! => {"changed": false, "msg": "Do not specify url_password in params as it may get logged"}

**Comment by Akasurde** [comment]
@uddhavb I guess this is totally different issue than original report. Could you please open a new issue with `-vvvv` logs and playbook used ? It will be easy for tracking purposes and will not hijack existing issue. Also, this repository is closed. Please open issue [here](https://github.com/ansible/ansible/issues/new)

**Comment by uddhavb** [comment]
@Akasurde yes, deleting the comment.

**Comment by Akasurde** [comment]
@uddhavb Thanks. Hope you have opened a new issue in new repo.

**Comment by uddhavb** [comment]
@Akasurde I found the solution in one of the issues posted here. So didn't need to post a new one. Thanks
